### PR TITLE
docs: add weekly report 0420

### PR DESCRIPTION
**Pull request by reeseliu715**
Add weekly report 0420

### CODE REVIEW

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/04/20/weekly-report-0420.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+![本周新晋贡献者](https://static.apiseven.com/202108/1650441612598-a27f8798-8f70-44e7-8c11-207173f776d4.png)
+
+## Good first issue
+
+### Issue #6803
+
+**链接**: https://github.com/apache/apisix/issues/6803
+
+**问题描述**：当用户在 Apache APISIX 中使用 openid-connect 插件时，使用了错误的 redirect_uri ，如下：
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/04/20/weekly-report-0420.md]:
```suggestion
**问题描述**：当用户在 Apache APISIX 中使用 `openid-connect` 插件时，使用了错误的 redirect_uri，如下：
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/04/20/weekly-report-0420.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+然后请求 "127.0.0.1:9080/"，将跳转到 500 页面，错误日志如下：
+
+![500页面](https://static.apiseven.com/202108/1650442371592-ad0c26f6-e3ab-49d8-a5cb-f2b033d87a91.png)
+
+```YAML
+2022/04/07 17:13:50 [error] 31780#3492140: *1959 [lua] openidc.lua:1378: authenticate(): request to the redirect_uri path but there's no session state found, client: 127.0.0.1, server: _, request: "GET / HTTP/1.1", host: "127.0.0.1:9080"
+2022/04/07 17:13:50 [error] 31780#3492140: *1959 [lua] openid-connect.lua:304: phase_func(): OIDC authentication failed: request to the redirect_uri path but there's no session state found, client: 127.0.0.1, server: _, request: "GET / HTTP/1.1", host: "127.0.0.1:9080"
+```
+
+这样的日志非常不清楚，并且用户不知道会发生什么。所以应该改进错误日志，以提高使用 openid-connect 插件的体验。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/04/20/weekly-report-0420.md]:
```suggestion
这样的日志非常不清楚，并且用户不知道会发生什么。所以应该改进错误日志，以提高使用 `openid-connect` 插件的体验。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/04/20/weekly-report-0420.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+### Issue #6797
+
+**链接**: https://github.com/apache/apisix/issues/6797
+
+**问题描述**：当使用 file-logger 插件时，可以通过定义 /dev/stdout 将日志发送到 stdout。这是为了使用 Docker 输出并将其在 Kubernetes 中传递到 ELK 堆栈。但是使用过程中出现了一个报错，表明当前用户的权限被拒绝。
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/04/20/weekly-report-0420.md]:
```suggestion
**问题描述**：当使用 `file-logger` 插件时，可以通过定义 /dev/stdout 将日志发送到 stdout。这是为了使用 Docker 输出并将其在 Kubernetes 中传递到 ELK 堆栈。但是使用过程中出现了一个报错，表明当前用户的权限被拒绝。
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/04/20/weekly-report-0420.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+## 近期功能特性亮点
+
+- [`key-auth` 支持隐藏认证相关的请求 header](https://github.com/apache/apisix/pull/6670)（贡献者：[bin-ya](https://github.com/bin-ya)）
+
+- [支持 Zookeeper 作为服务发现](https://github.com/apache/apisix/pull/6751)（贡献者：[shuaijinchao](https://github.com/shuaijinchao)）
```
Comment (SylviaBABY) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/04/20/weekly-report-0420.md]:
```suggestion
- [支持 ZooKeeper 作为服务发现](https://github.com/apache/apisix/pull/6751)（贡献者：[shuaijinchao](https://github.com/shuaijinchao)）
```

File: website/blog/2022/04/20/weekly-report-0420.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+**Issue description**:
+
+When using the `file-logger` plugin, it is possible to send the logging to stdout by defining `/dev/stdout`. This in order to use the docker output and relais this in kubernetes to a ELK stack. Though an error is thrown indicating a permission denied for the current user.failed to open file: `/dev/stdout`, error info: `/dev/stdout`: Permission denied while logging request
+
+## Highlights of Recent Features
+
+- [Add the function of hiding header for `key-auth` plugin](https://github.com/apache/apisix/pull/6670)(Contributor: [bin-ya](https://github.com/bin-ya))
+
+- [Support Zookeeper service discovery](https://github.com/apache/apisix/pull/6751)(Contributor: [shuaijinchao](https://github.com/shuaijinchao))
```
Comment (SylviaBABY) [inline website/blog/2022/04/20/weekly-report-0420.md]:
```suggestion
- [Support ZooKeeper as service discovery](https://github.com/apache/apisix/pull/6751)(Contributor: [shuaijinchao](https://github.com/shuaijinchao))
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/04/20/weekly-report-0420.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+![本周贡献者名单](https://static.apiseven.com/202108/1650441551212-28a65a2f-0d84-4f7d-a06e-fda4d072fff6.png)
+
+![本周新晋贡献者](https://static.apiseven.com/202108/1650441612598-a27f8798-8f70-44e7-8c11-207173f776d4.png)
+
+## Good first issue
+
+### Issue #6803
+
+**链接**: https://github.com/apache/apisix/issues/6803
```
Comment (guitu168) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/04/20/weekly-report-0420.md:34]:
```suggestion
**链接**：https://github.com/apache/apisix/issues/6803
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/04/20/weekly-report-0420.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+```YAML
+2022/04/07 17:13:50 [error] 31780#3492140: *1959 [lua] openidc.lua:1378: authenticate(): request to the redirect_uri path but there's no session state found, client: 127.0.0.1, server: _, request: "GET / HTTP/1.1", host: "127.0.0.1:9080"
+2022/04/07 17:13:50 [error] 31780#3492140: *1959 [lua] openid-connect.lua:304: phase_func(): OIDC authentication failed: request to the redirect_uri path but there's no session state found, client: 127.0.0.1, server: _, request: "GET / HTTP/1.1", host: "127.0.0.1:9080"
+```
+
+这样的日志非常不清楚，并且用户不知道会发生什么。所以应该改进错误日志，以提高使用 openid-connect 插件的体验。
+
+### Issue #6797
+
+**链接**: https://github.com/apache/apisix/issues/6797
```
Comment (guitu168) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/04/20/weekly-report-0420.md]:
```suggestion
**链接**：https://github.com/apache/apisix/issues/6797
```

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/04/20/weekly-report-0420.md (hunk 3 above)
Comment (guitu168) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/04/20/weekly-report-0420.md]:
```suggestion
**问题描述**：当使用 `file-logger` 插件时，可以通过定义 `/dev/stdout` 将日志发送到 stdout。这是为了使用 Docker 输出并将其在 Kubernetes 中传递到 ELK 堆栈。但是使用过程中出现了一个报错，表明当前用户的权限被拒绝。
```

File: website/blog/2022/04/20/weekly-report-0420.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+![500 Error](https://static.apiseven.com/202108/1650443701999-958f2096-d44d-4cd0-99b5-6e8833c361c6.png)
+
+```YAML
+
+2022/04/07 17:13:50 [error] 31780#3492140: *1959 [lua] openidc.lua:1378: authenticate(): request to the redirect_uri path but there's no session state found, client: 127.0.0.1, server: _, request: "GET / HTTP/1.1", host: "127.0.0.1:9080"
+2022/04/07 17:13:50 [error] 31780#3492140: *1959 [lua] openid-connect.lua:304: phase_func(): OIDC authentication failed: request to the redirect_uri path but there's no session state found, client: 127.0.0.1, server: _, request: "GET / HTTP/1.1", host: "127.0.0.1:9080"
+```
+
+This type of log is very unclear and the user does not know what to expect. Error logging should be improved to improve the experience of using the `Openid-connect` plug-in.
```
Comment (SylviaBABY) [inline website/blog/2022/04/20/weekly-report-0420.md:62]:
```suggestion
This type of log is very unclear and the user does not know what to expect. Error logging should be improved to improve the experience of using the `openid-connect` plug-in.
```

File: website/blog/2022/04/20/weekly-report-0420.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+## Good first issue
+
+### Issue #6803
+
+**Link**: https://github.com/apache/apisix/issues/6803
+
+**Issue description**:
+
+When I use `openid-connect` plugins with the wrong `redirect_uri` in Apache APISIX.Like flow:
```
Comment (SylviaBABY) [inline website/blog/2022/04/20/weekly-report-0420.md]:
```suggestion
When use `openid-connect` plugins with the wrong `redirect_uri` in Apache APISIX, as shown below:
```

File: website/blog/2022/04/20/weekly-report-0420.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+            "scope":"openid profile",
+            "bearer_only":false,
+            "introspection_endpoint_auth_method":"client_secret_post",
+            "redirect_uri":"http://127.0.0.1:9080/"
+             ...
+        }
+    },
+```
+
+Then, I request the "127.0.0.1:9080/", I will get 500, and the error log is as follow:
```
Comment (SylviaBABY) [inline website/blog/2022/04/20/weekly-report-0420.md]:
```suggestion
Then request the "127.0.0.1:9080/", will get 500, and the error log is as follow:
```

File: website/blog/2022/04/20/weekly-report-0420.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+
+![500 Error](https://static.apiseven.com/202108/1650443701999-958f2096-d44d-4cd0-99b5-6e8833c361c6.png)
+
+```YAML
+
```
Comment (guitu168) [inline website/blog/2022/04/20/weekly-report-0420.md]: del it

File: website/i18n/zh/docusaurus-plugin-content-blog/2022/04/20/weekly-report-0420.md (hunk 1 above)
Comment (guitu168) [inline website/i18n/zh/docusaurus-plugin-content-blog/2022/04/20/weekly-report-0420.md]:
```suggestion
**问题描述**：当用户在 Apache APISIX 中使用 openid-connect 插件时，使用了错误的 redirect_uri，如下：
```

File: website/blog/2022/04/20/weekly-report-0420.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+## Highlights of Recent Features
+
+- [Add the function of hiding header for `key-auth` plugin](https://github.com/apache/apisix/pull/6670)(Contributor: [bin-ya](https://github.com/bin-ya))
+
+- [Support Zookeeper service discovery](https://github.com/apache/apisix/pull/6751)(Contributor: [shuaijinchao](https://github.com/shuaijinchao))
+
+- [Feat(`request-id`): add algorithm support nanoid](https://github.com/apache/apisix/pull/6779)（Contributor: [aikin-vip](https://github.com/aikin-vip))
+
+- [Feat(`response-rewrite)`: support filters](https://github.com/apache/apisix/pull/6750)(Contributor: [kwanhur](https://github.com/kwanhur))
```
Comment (guitu168) [inline website/blog/2022/04/20/weekly-report-0420.md]:
```suggestion
- [Feat(`response-rewrite`): support filters](https://github.com/apache/apisix/pull/6750)(Contributor: [kwanhur](https://github.com/kwanhur))
```

**Review by bzp2010 — APPROVED** [review]

**Review by SylviaBABY — APPROVED** [review]

**Pull request merged**